filters/openpolicyagent/serveresponsewithregopolicy/serveresponsewithregopolicy.go

```diff
@@ -0,0 +1,105 @@
+package serveresponsewithregopolicy
+
+import (
+	"time"
+
+	"github.com/zalando/skipper/filters"
+	"gopkg.in/yaml.v2"
+
+	"github.com/zalando/skipper/filters/openpolicyagent"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/envoy"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/util"
+)
+
+type Spec struct {
+	factory openpolicyagent.OpenPolicyAgentFactory
+	opts    []func(*openpolicyagent.OpenPolicyAgentInstanceConfig) error
+}
+
+func NewServeResponseWithRegoPolicySpec(factory openpolicyagent.OpenPolicyAgentFactory, opts ...func(*openpolicyagent.OpenPolicyAgentInstanceConfig) error) Spec {
```

Comment:
return filters.Spec similar to https://github.com/zalando/skipper/pull/2407/files#diff-ef24f9c24cf4cebd7cb2fa58d0a0ca681aec47fcde32118c1d586b8ca4c965f3R20